### PR TITLE
feat: logging from step run executions

### PR DESCRIPTION
**Pull request by abelanger5**
# Description

Supports a simple logger up to 1000 lines from the UI, Typescript + Python SDKs. 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## What's Changed

- [X] Add logging to ingestor service
- [X] Add logging query to API
- [X] Add support to `ctx.log` from the TS SDK
- [X] Add support to `ctx.log` from the Py SDK

### CODE REVIEW

**Comment by steebchen** [comment]
note: please merge main into this branch to the bugfix for the failing load tests 🙏

File: frontend/app/src/pages/main/workflow-runs/$run/components/step-run-logs.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+      return 5000;
+    },
+  });
+
+  const logLines =
+    getLogsQuery.data?.rows?.reduce(
+      (acc, row) => acc + row.createdAt + ': ' + row.message + '\n',
+      '',
+    ) || 'No logs found';
```
Comment (grutt) [inline frontend/app/src/pages/main/workflow-runs/$run/components/step-run-logs.tsx:29]: once written maybe we should link to the docs for logs if the step ran with no logs

File: python-sdk/hatchet_sdk/clients/events.py [hunk 2]
```diff
@@ -28,10 +35,22 @@ def push(self, event_key, payload):
...
+    
+    def log(self, message: str, step_run_id: str):
+        try:
+            request = PutLogRequest(
+                stepRunId=step_run_id,
+                createdAt=proto_timestamp_now(),
+                message=message,
+            )
+
+            self.client.PutLog(request, metadata=get_metadata(self.token))
```
Comment (grutt) [inline python-sdk/hatchet_sdk/clients/events.py:54]: is this sync in python?

**Review by grutt — APPROVED** [review]

File: python-sdk/hatchet_sdk/clients/events.py (hunk 2 above)
Comment (abelanger5) [reply python-sdk/hatchet_sdk/clients/events.py:54]: Good catch. I'll use a thread pool executor for the event client logger.

File: frontend/app/src/pages/main/workflow-runs/$run/components/step-run-logs.tsx (hunk 1 above)
Comment (abelanger5) [reply frontend/app/src/pages/main/workflow-runs/$run/components/step-run-logs.tsx:29]: Yes

**Pull request merged**